Watchman.Discord/Areas/Protection/Strategies/DuplicatedMessagesDetectorStrategy.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+
+namespace Watchman.Discord.Areas.Protection.Strategies
+{
+    public class DuplicatedMessagesDetectorStrategy : ISpamDetector
+    {
+        public IUserMessagesCounter UserMessagesCounter { get; set; }
+
+        public DuplicatedMessagesDetectorStrategy(IUserMessagesCounter userMessagesCounter)
+        {
+            UserMessagesCounter = userMessagesCounter;
+        }
+
+        public SpamProbability GetSpamProbability(ServerMessagesCacheService serverMessagesCacheService, DiscordRequest request, Contexts contexts)
+        {
+            var userId = contexts.User.Id;
+            var lastFewMessages = serverMessagesCacheService.GetLastUserMessages(userId)
+                .TakeLast(4)
+                .ToList();
+
+            if (lastFewMessages.Count < 2)
+            {
+                return SpamProbability.None;
+            }
+
+            var content = request.OriginalMessage;
+            var similarMessagesCount = lastFewMessages.Count(x => GetDifferencePercent(x.Content, content) < 0.4);
```

Comment:
ustawiłbym wysoki procent typu +90%
bo przy dłuższych wiadomościach niż kilka znaków lubi walić duże wyniki do (dla człowieka) kompletnie innych tekstów

Reply:
w sensie 
GetDifferencePercent(x.Content, content) < 0.1 
czy GetDifferencePercent(x.Content, content) < 0.9
?
bo jeśli to drugie to przy krótkich wiadomościach prawie zawsze wykrywa spam

Reply:
hmm
to może być problem
ale to też może być zaleta 

bo teoretycznie to możemy chcieć żeby ktoś napisał jedną długą wiadomość
a nie 20 wiadomości po jednym słowie 

jako krótką wiadomość masz na myśli jedno zdanie czy jedno słowo?

Reply:
mam na myśli te przykładowe wiadomości w testach

Reply:
no jeśli ktoś będzie walił dużo takich drobnych wiadomości w krótkim czasie
zamiast jednej
to chyba dobrze że wykryje jako spam